src/classes/ALLO_PaymentSync_TEST.cls

```diff
@@ -832,6 +832,78 @@ private with sharing  class ALLO_PaymentSync_TEST {
         Test.stopTest();
     }
 
+    /*******************************************************************************************************
+    * @description Test when an opportunity is fully scheduled with one paid payment and one scheduled payment,
+    * updating the allocations should not change the paid payment allocation. However, it should update the
+    * schedule payment's allocations.
+    ********************************************************************************************************/
+    @isTest
+    private static void testSyncOpportunityToPaymentWithFullyScheduledPartiallyPaid() {
+        setupForTests();
+        makeGAUs(1);
+        opportunity = [SELECT Id, Name, Amount FROM Opportunity LIMIT 1];
+        List<Allocation__c> alloForInsert = new List<Allocation__c>();
+
+        alloForInsert.add(new Allocation__c(Opportunity__c=opportunity.Id, Percent__c = 100, General_Accounting_Unit__c = gaus[0].Id));
+        insert alloForInsert;
+
+        makePayments(1000, 2);
+
+        npe01__OppPayment__c paidPayment = new npe01__OppPayment__c(Id = payments[0].Id, npe01__Paid__c = true);
+        update paidPayment;
+
+        Allocation__c oppAllocation = [select Id,
+                Opportunity__c,
+                Payment__c,
+                General_Accounting_Unit__c,
+                General_Accounting_Unit__r.Name,
+                Amount__c,
+                Percent__c
+        from Allocation__c
+        where Opportunity__c = :opportunity.Id
+        AND General_Accounting_Unit__c =: gaus[0].Id
+        order by Amount__c DESC];
+
+        oppAllocation.Percent__c = 50;
+
+        Test.startTest();
+        update oppAllocation;
+
+        List<Allocation__c> paidPaymentAllocationList = new List<Allocation__c>([select Id,
+                Opportunity__c,
+                Payment__c,
+                Payment__r.npe01__Paid__c,
+                General_Accounting_Unit__c,
+                General_Accounting_Unit__r.Name,
+                Amount__c,
+                Percent__c
+        from Allocation__c
+        where Payment__c = :payments[0].Id
+        order by Amount__c DESC]);
+
+        System.assertEquals(1, paidPaymentAllocationList.size());
```

Comment:
Should still have the third part. 'Paid Payment Allocation Count' 'Paid Payment Amount', etc.  It just doesn't need to have the expected value as part of it.  E.g.
System.assertEquals(1, paidPaymentAllocationList.size(), 'Payment Payment Allocation Count');

Reply:
Updated